lightning/src/ln/outbound_payment.rs

```diff
@@ -158,6 +176,12 @@ impl PendingOutboundPayment {
 				payment_hash: *payment_hash,
 				reason: Some(reason)
 			};
+		} else if let PendingOutboundPayment::InvoiceReceived { payment_hash, .. } = self {
```

Comment:
Re: `ChannelManager::abandon_payment` -- it currently doesn't have an effect on payments where we're `AwaitingInvoice`. Would it be possible to make the `payment_hash` optional in `PendingOutboundPayment::Abandoned` and generate an `InvoiceRequestFailed` when the user calls `abandon_payment`?

Reply:
Missed this but working on it now. Should we add a reason to `InvoiceRequestFailed` like `PaymentFailed` now that it may not simply be because of a timeout?

Reply:
Note that it is a little weird as `ChannelManager::abandon_payment` would call `OutboundPayments::abandon_payment` with `PaymentFailureReason::UserAbandoned`. We would either ignore the reason for one specific to `InvoiceRequestFailed` or reuse it, even though all the variants are not possible and would require another variant for timeout. For now, I've left a reason out.

Reply:
That seems reasonable but seems fine to leave for follow-up too